# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../fam/gcc43.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./lib/Client.c++
+++ ./lib/Client.c++
@@ -34,7 +34,7 @@
 #include <syslog.h>
 #include <errno.h>
 
-#include <iostream.h>
+#include <iostream>
 
 #include "fam.h"
 #include "Client.h"
--- ./src/DNotify.c++
+++ ./src/DNotify.c++
@@ -31,6 +31,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <libgen.h>
+#include <cstdlib>
 
 #include "DNotify.h"
 
--- ./src/IMon.c++
+++ ./src/IMon.c++
@@ -40,7 +40,7 @@
 #include "Interest.h"
 #include "Log.h"
 #include "Scheduler.h"
-#include "alloc.h"
+#include <memory>
 
 int		   IMon::imonfd = -2;
 IMon::EventHandler IMon::ehandler = NULL;
--- ./src/DNotify.h.old	2008-08-25 15:27:30.000000000 +0200
+++ ./src/DNotify.h	2008-08-25 15:30:19.000000000 +0200
@@ -59,8 +59,8 @@
     static int pipe_write_fd;
     static int pipe_read_fd;
     static int change_queue[QUEUESIZE];
-    static volatile sig_atomic_t DNotify::queue_overflowed;
-    static volatile sig_atomic_t DNotify::queue_changed;
+    static volatile sig_atomic_t queue_overflowed;
+    static volatile sig_atomic_t queue_changed;
     static volatile int queue_head; // Only modified by read handler
     static volatile int queue_tail; // Only modified by signal handler
     static EventHandler ehandler;
